.github/workflows/build.yml

```diff
@@ -0,0 +1,121 @@
+name: Build
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  prep:
+    name: Prepare the build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Update the 'latest' tag
+        if: github.ref == 'refs/heads/main'
+        run: |
+           git config user.name github-actions
+           git config user.email github-actions@github.com
+           git push origin :refs/tags/latest || true
+           git tag -m latest -fa latest || true
+           git push origin main --tags
+  build:
+    name: Build on ${{ matrix.os }} with Java ${{ matrix.java }}
+    needs: prep
```

Comment:
I think this is probably OK, because you have the conditional on the step in the other job, but if you made the whole job conditional, it would likely stall

Reply:
> I think this is probably OK, because you have the conditional on the step in the other job, but if you made the whole job conditional, it would likely stall

OK